dynamic_instrumentation/src/instrument_memory.rs

```diff
@@ -465,148 +517,154 @@ impl<'a, 'tcx: 'a> Visitor<'tcx> for CollectFunctionInstrumentationPoints<'a, 't
+                                location,
+                                load_value_fn,
+                                vec![InstrumentationOperand::from_type(
+                                    Operand::Copy(dest),
+                                    &dest_ty,
+                                )],
+                                false,
+                                false,
+                                EventMetadata {
+                                    source: None,
+                                    destination: Some(to_mir_place(&dest)),
+                                    transfer_kind: TransferKind::None,
+                                },
+                            );
+                        } else {
+                            // Copying a pointer to another place; trace the destination
+                            self.add_instrumentation_point(
+                                location,
+                                copy_fn,
+                                vec![InstrumentationOperand::from_type(
+                                    Operand::Copy(dest),
+                                    &dest_ty,
+                                )],
+                                false,
+                                false,
+                                EventMetadata {
+                                    source: Some(to_mir_place(p)),
+                                    destination: Some(to_mir_place(&dest)),
+                                    transfer_kind: TransferKind::None,
+                                },
+                            );
+                        }
                     }
-                } else if let Rvalue::Cast(_, p, _) = value {
-                    location.statement_index += 1;
-                    if p.ty(&self.body.local_decls, self.tcx).is_integral() {
-                        // Casting integer to a pointer; trace the destination
-                        self.add_instrumentation_point(
-                            location,
-                            ptr_contrive_fn,
-                            vec![InstrumentationOperand::RawPtr(Operand::Copy(dest))],
-                            false,
-                            false,
-                            EventMetadata {
-                                source: p.place().as_ref().map(to_mir_place),
-                                destination: Some(to_mir_place(&dest)),
-                                transfer_kind: TransferKind::None,
-                            },
-                        );
-                    } else {
-                        // Casting between pointer types; trace the destination
+                    Rvalue::Cast(_, p, _) => {
+                        location.statement_index += 1;
+                        if p.ty(&self.body.local_decls, self.tcx).is_integral() {
+                            // Casting integer to a pointer; trace the destination
+                            self.add_instrumentation_point(
```

Comment:
This call could also be combined with the one below pretty cleanly.

Reply:
will do in #552